src/main/java/org/apache/sysds/runtime/instructions/fed/AggregateBinaryFEDInstruction.java

```diff
@@ -53,6 +56,12 @@ public AggregateBinaryFEDInstruction(Operator op, CPOperand in1, CPOperand in2,
 		super(FEDType.AggregateBinary, op, in1, in2, out, opcode, istr, fedOut);
 	}
 	
+	public AggregateBinaryFEDInstruction(AggregateBinaryCPInstruction instruction) {
+		// TODO @KEVIN: should we update the instruction string?
+		this(instruction.getOperator(), instruction.input1, instruction.input2, instruction.output,
+			instruction.getOpcode(), instruction.getInstructionString(), FederatedOutput.NONE);
```

Comment:
I always keep the instruction string without changes, but we might want to replace the leading `CP` with `FED` and add the type of `FederatedOutput` (always `NONE`, as it is decided at runtime).